rules/framework.bzl

```diff
@@ -162,7 +162,15 @@ def _framework_packaging(ctx, action, inputs, outputs, manifest = None):
     if action in ctx.attr.skip_packaging:
         return []
     action_inputs = [manifest] + inputs if manifest else inputs
-    outputs = [ctx.actions.declare_file(f) for f in outputs]
```

Comment:
Minor: To avoid 1 iteration - can you do something like:
```
    outputs = [ctx.actions.declare_directory(f) if f.is_directory else ctx.actions.declare_file(f) for f in outputs]
```

Reply:
Happy to collapse into a single loop. Was having some issues with this differing between Bazel 5 and Bazel 6 so seeing if it passes on CI then will update. 👍

Reply:
And in this particular case, the `symlink` action is complaining about the input `File` from `inputs` not matching the declared so it'll need to be a loop over indexes anyway. Something like:
```Python
    outputs = [
        ctx.actions.declare_directory(outputs[index]) if inputs[index].is_directory else ctx.actions.declare_file(outputs[index])
        for index in range(len(outputs))
    ]
```
Let me know if this isn't okay with you, @jerrymarino.

Reply:
Cool, makes sense! Yeah perhaps either Bazel 6 or rule bumps ended up getting us into a situation that converted something that was a file into dir?

Reply:
The case the `rules_ios` test suite catches is the intent definition's declaration of `.h`/`.m` directories:

https://github.com/bazelbuild/rules_apple/blob/c13b1072832737c7ca7d24ec598a53c590e4e010/apple/internal/resource_rules/apple_intent_library.bzl#L55-L56